### ISSUE TITLE
Is the workflow being overwritten and saved?

### ISSUE THREAD

**Issue by nefudev** [11cafe/comfyui-workspace-manager#68]
When I change a workflow saved by comfyui-workspace-manager without saving it and refresh the browser, it is saved. Is this a specification? I would like the workflow to be saved when the save button is pressed.

**Environment**
comfy:2c4e92a98b8338f754855a0db7dce164945e366e
this node:faec53571df6c91a6b7ad9813f208a0210d3a5c0
browser:Chrome/firefox

1. Create a workflow, save it, and execute it.
  In this example, a LoadImage and a Preview are placed and named "img_A".
![img_001](https://github.com/11cafe/comfyui-workspace-manager/assets/132149510/24fc0e9f-cb5a-4ee0-8da0-38e150821015)

2. Modify the "img_A" workflow saved above without saving it.
　Here, we have replaced the loaded image and added a Preview.
![img_002](https://github.com/11cafe/comfyui-workspace-manager/assets/132149510/c2a79fe1-0569-4b47-bd35-acff70502ee5)

3. Restart the browser or restart ComfyUI and select "img_A" to see the modified workflow. Saved img_A can no longer be displayed.
![img_003](https://github.com/11cafe/comfyui-workspace-manager/assets/132149510/012b4e5d-4b55-41a8-8718-748bdd9baa61)

**Comment by Weixuanf** [comment]
Hi yes right now it is auto saving your workflow, this is to avoid accidentally losing your unsaved changes, (for example, you accidentally refreshed your page without pressing saving can cause losing your valuable changes)

You can easily discard the unsaved changes and revert to your last saved status by pressing "Discard unsaved changes":

<img width="449" alt="Screenshot 2024-01-03 at 7 35 45 PM" src="https://github.com/11cafe/comfyui-workspace-manager/assets/18367033/11ab56bb-9cf2-4897-aa61-be8ffbd0896a">

I chose this strategy on purpose because this not only avoids losing any changes accidentally but also lets you go back to your last manually saved version easily. So you have the best of two worlds. You also gets a automatic version history too! Everytime you press save, it logs a new version in the history so you can easily revert back to any history versions you have!

**Comment by nefudev** [comment]
Thank you for your answer.

I was not aware of the auto-save and "Discard unsaved changes" features.
I checked the "Discard unsaved changes" feature and sometimes it works and sometimes it doesn't...

I think the auto-save and "Discard unsaved changes" are very useful for experimental workflows.However, I personally want to create a template workflow for each function and do not want to make any changes in the first place. Therefore, I would like the ability to lock the workflow to read-only.

For example, if I select a locked workflow, it will create a new workflow with that workflow.

**Comment by Weixuanf** [comment]
Ok I see, i think what you want is a "Lock" button to lock a certain workflow once locked it cannot be modified until unlocked.

> I checked the "Discard unsaved changes" feature and sometimes it works and sometimes it doesn't...
when it doesn't work, try the "Version history" option on a workflow, it will list all the "snapshot json" everytime you pressed save on this workflow. you can easily go back to any version you like by pressing it. so you will never lose any versions or changes.

<img width="387" alt="Screenshot 2023-12-29 at 11 17 37 PM" src="https://github.com/11cafe/comfyui-workspace-manager/assets/18367033/ac8c52d7-c339-46a0-8682-ed03b0cab75d">

**Comment by nefudev** [comment]
Thank you! Looking forward to it.
I checked the functionality in the version history!

**Comment by gisu** [comment]
Please Remove Auto Save by Default. I have quite a few problems with this, workflow files with .json.json.json.json. Furthermore, I call workflows in the manager and often get a different workflow. This is probably because I have ComfyUI running on multiple tabs, which messes up the manager.
Therefore it would be nice if I could switch on or off Auto Save in the settings.

**Comment by Weixuanf** [comment]
> Please Remove Auto Save by Default. I have quite a few problems with this, workflow files with .json.json.json.json. Furthermore, I call workflows in the manager and often get a different workflow. This is probably because I have ComfyUI running on multiple tabs, which messes up the manager. Therefore it would be nice if I could switch on or off Auto Save in the settings.

Yes! We are working on disabling workflow auto save completely and enabling multi tab browsing! 
right now the multiple browser tab experience is not good, I agree I may sometimes overwrite each other if you have multiple browser tabs open.

stay tuned! Will update here when launched

**Comment by pvp-by** [comment]
https://github.com/11cafe/comfyui-workspace-manager/commit/51084bb0389f22c538cd301ae114903fb768aa92
The version of the extension is up to date at the time of my post. 
I attach the settings I have (they are default, I have not changed anything):
![image](https://github.com/11cafe/comfyui-workspace-manager/assets/10133995/8b62ecb9-b062-4e0c-8df6-897dd6dc9971)

but if I refresh the page, I see a warning about unsaved, (I don't click yes/no) the page reloads and I get the default workflow (which is load default) no undo version in the list, but my workflow is overwritten.

If it helps you, the comfyUI server is on a neighboring computer relative to the browser tab where I am. 
time `22-49` (~8 min change last data - deleted)
![image](https://github.com/11cafe/comfyui-workspace-manager/assets/10133995/bb2c057b-bf5e-4d87-85ba-b31216053d5a)
